### DISCUSSION TITLE
[Bug] Persisting is so long

### DISCUSSION THREAD

**Discussion by aliuq** [ScoopInstaller/Scoop#5455] [Help and Questions]
#### Current Behavior

Hi

During the installation process of Scoop, it seems to be stuck on the "persist data" step. I have many applications installed globally, but only a few applications installed locally. I have noticed that the applications installed in the local directory finish installation quickly, and I have confirmed that the "persist data" for the installed applications already exist. However, the process is still not terminated, and I am not sure if it's still processing other issues.

It tooks 7 minutes to complete the installation.

The global apps count is 70

![image](https://user-images.githubusercontent.com/11707934/226264234-ff212eba-1ab1-4df3-854f-e581383d5cd3.png)

![image](https://user-images.githubusercontent.com/11707934/226266947-cbc807e2-cb00-4625-a0bf-8ebd0f61ead9.png)

#### Expected Behavior

Install quickly

### System details

**Windows version:** `11`

**OS architecture:** `64bit`

**PowerShell version:** `7.3.2`

**Additional software:** None

#### Scoop Configuration

```json
{
  "aria2-enabled": false,
  "get": "debug",
  "alias": {
    "rmrf": "scoop-rmrf",
    "ls": "scoop-ls"
  },
  "last_update": "2023-03-20T14:29:21.8841773+08:00",
  "scoop_repo": "https://github.com/lukesampson/scoop",
  "scoop_branch": "master"
}
```

**Comment by rashil2000** [comment]
How large is the resources/lsp folder? 

Does this issue still exist if the global scoop directory is set to some other location?

**Reply by aliuq** [reply]
I have observed that the resources/lsp folder is very small and when installed in the user directory, it only takes a few seconds. However, when installing or updating the application globally, the persisting time is significantly longer. I suspect this is due to the persist folder being very large, with 36GB of data and 100w files.

![image](https://user-images.githubusercontent.com/11707934/229498473-c607acef-9e3c-430e-8781-a224cbf3bee1.png)

**Reply by rashil2000** [reply]
Hmm, I see. I assume the user persist directory is quite small compared to the global one?

**Reply by aliuq** [reply]
Yes, that's it.

**Reply by rashil2000** [reply]
Hmm, it's a strange issue, I haven't worked with folders that big in my testing, sorry.

**Comment by Aetylus** [comment]
Running into the same issue, with a persist directory size of approximately 1.27 GB - all of which is due to the persist directory associated with a nodejs installation.

After removing nodejs and deleting it's persist directory, the installation of another package that persists a directory (such as GPG) goes as expected.

I think for now I'll just avoid installing nodejs through scoop, but I am curious why persisting is so negatively impacted by another package's persist size.
